commands/alk.js

```diff
@@ -0,0 +1,66 @@
+const embedService = require("../services/embedService");
+const db = require("../models").sequelize;
+const userModel = db.models.User;
+const mediaModel = db.models.Media;
+const mime = require("mime-types");
+const axios = require("axios").default;
+
+const getIntention = (msg, args) =>
+  args && args[0] === "add" && args[1]
+    ? addResource(msg, args)
+    : getResource(msg, args);
+
+const URLExists = async (url) => {
+  try {
+    await axios.head(url);
+    return true;
+  } catch {
+    return false;
+  }
+};
+
+const checkURL = (url) => {
+  const pattern = /(ftp|http|https):\/\/(\w+:{0,1}\w*@)?(\S+)(:[0-9]+)?(\/|\/([\w#!:.?+=&%@!\-\/]))?/;
+  return pattern.test(url);
+};
+
+const checkResource = async (resource) => {
+  const exists = await URLExists(resource);
+  if (checkURL(resource) && exists) {
+    return mime.lookup(resource);
+  }
+};
+
+const addResource = async (msg, args) => {
+  const user = await userModel.findOrCreateByDiscordId(msg.author.id);
+  const passedResource = args[1];
+  const resourceExists = await mediaModel.findOneByMediaContent(passedResource);
+  const resourceValid = await checkResource(passedResource);
+  if (user.whitelisted && resourceValid && !resourceExists) {
```

Comment:
Move `user.whitelisted` up a few lines so we don't have to await the other calls

Reply:
Fixed.